pvlib/pvsystem.py

```diff
@@ -2844,22 +2841,25 @@ def singlediode(photocurrent, saturation_current, resistance_series,
        photovoltaic cell interconnection circuits" JW Bishop, Solar Cell (1988)
        https://doi.org/10.1016/0379-6787(88)90059-2
     """
+    if ivcurve_pnts:
```

Comment:
Probably should add a test in `test_pvsystem` to check that this warning is emitted, if nothing else, to ensure coverage of these lines.